analyzers/src/SonarAnalyzer.Common/Rules/CertificateValidationCheckBase.cs

```diff
@@ -142,19 +142,19 @@ protected ImmutableArray<Location> BlockLocations(InspectionContext c, SyntaxNod
             return ret.ToImmutable();
         }
 
-        protected virtual SyntaxNode FindRootClassOrRecordOrModule(SyntaxNode node)
+        protected virtual SyntaxNode FindRootTypeDeclaration(SyntaxNode node)
         {
             SyntaxNode candidate;
-            var current = node.FirstAncestorOrSelf<SyntaxNode>(IsClassOrRecordDeclaration);
-            while (current != null && (candidate = current.Parent.FirstAncestorOrSelf<SyntaxNode>(IsClassOrRecordDeclaration)) != null)  // Search for parent of nested class/record
+            var current = node.FirstAncestorOrSelf<SyntaxNode>(IsTypeDeclaration);
+            while (current != null && (candidate = current.Parent.FirstAncestorOrSelf<SyntaxNode>(IsTypeDeclaration)) != null)  // Search for parent of nested class/record
```

Comment:
The case when current is null is not covered for this line. I will postpone this for now since this can be easily covered once we support top level statements.